libvast/include/vast/plugin.hpp

```diff
@@ -471,19 +480,24 @@ class plugin_ptr final {
 
 namespace vast::plugins {
 
-template <typename Plugin>
-const Plugin* find(const std::string& name) {
+template <class Plugin>
+const Plugin* find(std::string_view name) noexcept {
   const auto& plugins = get();
-  auto it
+  const auto found
     = std::find_if(plugins.begin(), plugins.end(), [&](const plugin_ptr& p) {
-        return name == p->name();
+        const auto plugin_name = p->name();
+        return std::equal(name.begin(), name.end(), plugin_name.begin(),
+                          plugin_name.end(),
+                          [](unsigned char lhs, unsigned char rhs) noexcept {
+                            return std::tolower(lhs) == std::tolower(rhs);
+                          });
```

Comment:
Nit: these seem so generally useful that I would almost put them as helper functions into `detail/strings.hpp`. Having something like `to_upper` and `to_lower` that can be invoked on a string(-view) would make code a lot more readable.

Reply:
I chose to make plugin_ptr comparable with a strong ordering instead; this removes the need for the custom compare object in three places.